packages/eas-cli/src/build/runBuildAndSubmit.ts

```diff
@@ -208,7 +134,12 @@ export async function runBuildAndSubmitAsync(
       flags,
       moreBuilds: platforms.length > 1,
       buildProfile,
-      resourceClass: resolveBuildResourceClass(buildProfile, flags.resourceClass),
+      resourceClass: await resolveBuildResourceClassAsync(
+        buildProfile,
+        projectDir,
+        getDynamicProjectConfigAsync,
```

Comment:
do not call getDynamicProjectConfigAsync more than once if it's not necessary, just resolve resouce class latter when you have exp